### PR TITLE
koord-scheduler: fix device cache potential concurrency issue

### PR DESCRIPTION
**Pull request by jasonliu747**
### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/scheduler/plugins/deviceshare/device_cache.go [hunk 1]
```diff
@@ -382,15 +382,20 @@ func newNodeDeviceCache() *nodeDeviceCache {
...
+	// getNodeDevice will call createNodeDevice(nodeName) if needInit is true and nodeDeviceInfos[nodeName] is nil
+	if n.nodeDeviceInfos[nodeName] == nil && needInit {
+		klog.V(5).Infof("node device cache not found, nodeName: %v, createNodeDevice", nodeName)
+		return n.createNodeDevice(nodeName)
+	}
+
 	return n.nodeDeviceInfos[nodeName]
 }
 
 func (n *nodeDeviceCache) createNodeDevice(nodeName string) *nodeDevice {
```
Comment (eahydra) [inline pkg/scheduler/plugins/deviceshare/device_cache.go:398]: It is recommended to remove this function and merge with getNodeDevice.

File: pkg/scheduler/plugins/deviceshare/device_cache.go [hunk 2]
```diff
@@ -382,15 +382,20 @@ func newNodeDeviceCache() *nodeDeviceCache {
 	}
 }
 
-func (n *nodeDeviceCache) getNodeDevice(nodeName string) *nodeDevice {
-	n.lock.RLock()
-	defer n.lock.RUnlock()
+func (n *nodeDeviceCache) getNodeDevice(nodeName string, needInit bool) *nodeDevice {
+	n.lock.Lock()
```
Comment (eahydra) [inline pkg/scheduler/plugins/deviceshare/device_cache.go:386]: Please change the lock type to sync.Mutex

**Review by eahydra — CHANGES_REQUESTED** [review]

File: pkg/scheduler/plugins/deviceshare/device_cache.go (hunk 1 above)
Comment (jasonliu747) [reply pkg/scheduler/plugins/deviceshare/device_cache.go]: sure, no problem.

File: pkg/scheduler/plugins/deviceshare/device_cache.go (hunk 2 above)
Comment (jasonliu747) [reply pkg/scheduler/plugins/deviceshare/device_cache.go:386]: changed to sync.Mutex

**Review by eahydra — APPROVED** [review]
/lgtm
/approve

**Review by jasonliu747 — COMMENTED** [review]
/approve

**Pull request merged**